src/core/src/index/revindex/disk_revindex.rs

```diff
@@ -172,6 +173,12 @@ impl DiskRevIndex {
         }))
     }
 
+    /// Access to the DB is unsafe because RocksDB allows writing with
+    /// a regular Arc<DB> handle; it doesn't have to be mut.
+    pub unsafe fn db(&self) -> &Arc<DB> {
+        &self.db
+    }
```

Comment:
I think this can be a regular `Arc<DB>`, and we return a cheap clone (refcount increase):

```suggestion
    pub unsafe fn db(&self) -> Arc<DB> {
        self.db.clone()
    }
```